synapse/rest/client/v1/login.py

```diff
@@ -129,6 +131,21 @@ def on_OPTIONS(self, request):
 
     @defer.inlineCallbacks
     def on_POST(self, request):
+
+        time_now = self.hs.clock.time()
+
+        allowed, time_allowed = self._address_ratelimiter.can_do_action(
+            request.getClientIP(), time_now_s=time_now,
+            rate_hz=self.hs.config.rc_login_requests_per_address_per_second,
```

Comment:
it would be nice to see these config settings pulled out into member fields in the constructor.